lib/addons/src/types.ts

```diff
@@ -253,7 +253,7 @@ export interface BaseMeta<ComponentType> {
 }
 
 export interface BaseStory<Args, StoryFnReturnType> {
-  (args: Args, context: StoryContext): StoryFnReturnType;
+  (args?: Args, context?: StoryContext): StoryFnReturnType;
```

Comment:
We tend to use our stories in tests too, so having `args` and `context` as required arguments for stories makes this difficult..

```ts
import { module, test } from 'qunit'
import { render } from '@ember/test-helpers'
import { GlimmerButton } from '../stories/ember-typescirpt.stories'

module('Integration | Component | button', function (hooks) {
  test('some test', async function (assert) {
	// If `args` and `context` are required here 👇 that makes tests difficult to write
    const { context, template } = GlimmerButton(/* args, context */)
    this.setProperties(context)

    await render(template)

    assert.ok('some assertion')
  })
})
```

Open to alternatives 👀

Reply:
`(this.)setProperties()` is a ember code that is going into extinction. While this is still working, think about alternatives, maybe:


```
this = { this, ...context };
```

I don't know if that even works 😂

Reply:
Never seen this kind of test writing before. Never did it myself. Interessting though 👍

Reply:
I haven't seen anything mentioning the removal of `setProperties` :eyes:

This is something provided by https://github.com/emberjs/ember-qunit#setup-tests

If it does go away, `Object.assign(this, context)` would do the job

> Never seen this kind of test writing before.

We find it pretty useful. It allows us to use our stories for testing. We can setup some story for our app in storybook, and write a test that verifies that story 🎉

Reply:
Just today, wrapped the above code into a helper function

```ts
test('some test', async function (assert) {
  await renderStory(this, GlimmerButton, { emoji: '⚽️' })
  assert.ok('some assertion')
})
```

That wraps the rendering of the story template and passing the arguments..

```js
import { render } from '@ember/test-helpers'

export async function renderStory(testContext, story, args) {
  const storyArgs = { ...story.args, ...args }
  const { context, template } = story(storyArgs)
  Object.assign(testContext, context)
  return render(template)
}
```

Reply:
this is neat 👍 

well `this.set` and `this.get` are transferred over from classic/legacy ember. You don't need them any longer (at least for initial values). I think, changing this post-rendering still requires `this.set`. At least personally I like to get rid of 😉